shared/js/smart-screen/card_store.js

```diff
@@ -104,10 +103,51 @@
           resolve();
         }
       });
+    },
+
+    iterateData: function cs_listData(cb) {
+      return new Promise((resolve, reject) => {
+
+        this._getStore().then(store => {
+          return store.sync();
+        }).then(function iterateTask(cursor) {
+          cursor.next().then(task => {
+            if (task.operation == 'add') {
+              cb(task.data, task.id);
+              iterateTask(cursor);
+            } else if (task.operation == 'clear') {
```

Comment:
Yeah, but since the cursor runs as 'clear'-'add'-'add'-....'done' for a new cursor, so I hope to keep it to remind the sequence looks like that.
I'll add some comments for that.